### PR TITLE
[PackageTool] Add package edit subtool

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: Sources/Commands/SwiftPackageTool.swift [hunk 1]
```diff
@@ -197,6 +223,29 @@ public class SwiftPackageTool: SwiftTool<PackageMode, PackageToolOptions> {
...
+            let manifests = try workspace.loadDependencyManifests()
+            // Look for the package's manifest.
+            guard let manifest = manifests.lookup(packageName) else {
+                throw PackageToolOperationError.packageNotFound
+            }
+            guard let dependency = workspace.dependencyMap[RepositorySpecifier(url: manifest.url)] else {
+                fatalError("Unexpected failure, dependency for \(manifest.url) not found in workspace.")
+            }
+            // Create revision object if provided by user.
+            let revision = options.editRevision.flatMap(Revision.init(identifier:))
```
Comment (ddunbar) [inline Sources/Commands/SwiftPackageTool.swift]: `{ Revision(identifier: $0) }` seems just as clear?

File: Sources/Commands/SwiftPackageTool.swift [hunk 2]
```diff
@@ -264,6 +313,14 @@ public class SwiftPackageTool: SwiftTool<PackageMode, PackageToolOptions> {
         print("")
         print("NOTE: Use `swift build` to build packages, and `swift test` to test packages")
     }
+
+    var editUsage: String {
+        let stream = BufferedOutputByteStream()
+        stream <<< "Expected package edit format:\n"
+        stream <<< "swift package edit --name <packageName> (--revision <revision> | --branch <newBranch>)\n"
```
Comment (ddunbar) [inline Sources/Commands/SwiftPackageTool.swift]: By convention, we should use `[--revision <revision>` to connote optional arguments.

File: Sources/Commands/Workspace.swift [hunk 3]
```diff
@@ -615,7 +617,7 @@ public class Workspace {
                 // FIXME: Nevertheless, we should handle this failure explicitly.
                 //
                 // FIXME: We should have a cache for this.
-                let manifest: Manifest = try! manifestLoader.load(packagePath: checkoutsPath.appending(managedDependency.subpath), baseURL: managedDependency.repository.url, version: managedDependency.currentVersion)
+                let manifest: Manifest = try! manifestLoader.load(packagePath: packagePathBase.appending(managedDependency.subpath), baseURL: managedDependency.repository.url, version: managedDependency.currentVersion)
```
Comment (ddunbar) [inline Sources/Commands/Workspace.swift]: We should have test coverage of this?

**Comment by ddunbar** [comment]
We should have test coverage for the tool behaviors.

File: Sources/Commands/SwiftPackageTool.swift (hunk 2 above)
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftPackageTool.swift]: I wasn't sure I used this for reference http://docopt.org which says `(A|B)` is either/or which is more fitting here

Comment (ddunbar) [inline Sources/Commands/SwiftPackageTool.swift]: oh, right, I forgot we decided to require you to specify one.

File: Sources/Commands/Workspace.swift (hunk 3 above)
Comment (aciidb0mb3r) [inline Sources/Commands/Workspace.swift]: right, will add

**Comment by aciidb0mb3r** [comment]
functional?

**Comment by ddunbar** [comment]
I think so, yeah? Doesn't have to try to do everything, but I think it should vet the behaviors.

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**